python/proton/_common.py

```diff
@@ -80,12 +82,16 @@ def unicode2utf8(string):
 
 def utf82unicode(string):
     """Convert C strings returned from proton-c into python unicode"""
-    if string is None:
+    if string is None or string == ffi.NULL:
         return None
     elif isinstance(string, unicode):
         # py2 unicode, py3 str (via hack definition)
         return string
```

Comment:
something you'd be able to revisit in another PR down the road; the minimal Python version you need to support is Python 3.6, so any hacks for Python 2 should be deleted from the code